sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java

```diff
@@ -49,22 +49,22 @@ public void validateValueType(int index, Object fieldValue)
     * Get the coder for {@link BeamRecordCoder}.
     */
    public BeamRecordCoder getRecordCoder(){
-     return BeamRecordCoder.of(this, fieldsCoder);
+     return BeamRecordCoder.of(this, fieldCoders);
    }
 
-   public List<String> getFieldsName(){
-     return fieldsName;
+   public List<String> getFieldNames(){
```

Comment:
field doesn't equal with field name, which happens to be true here. For example, _field_ consists of field name and field type in sql.